### PR TITLE
Use singleton type over type member

### PR DESCRIPTION
**Pull request by afsalthaj**

### CODE REVIEW

File: zio-http/src/main/scala/zio/http/api/ServiceSpec.scala [hunk 1]
```diff
@@ -48,8 +48,8 @@ object ServiceSpec                        {
     mo: Combiner.WithOut[MO1, MO2, MO3],
   ) extends ServiceSpec[MI3, MO3, AllIds]
 
-  def apply[A <: EndpointSpec[_, _]](api: A): ServiceSpec[Unit, Unit, api.Id] =
-    Single(api.asInstanceOf[EndpointSpec.WithId[api.Id, Any, Any]])
+  def apply[Id](api: EndpointSpec[_, _]): ServiceSpec[Unit, Unit, Id] =
```
Comment (jdegoes) [inline zio-http/src/main/scala/zio/http/api/ServiceSpec.scala]: This is unconstrained phantom type and therefore unsafe: can you do api.type in return value instead??

Comment (jdegoes) [reply zio-http/src/main/scala/zio/http/api/ServiceSpec.scala]:
```suggestion
  def apply(api: EndpointSpec[_, _]): ServiceSpec[Unit, Unit, api.type] =
```

Comment (afsalthaj) [reply zio-http/src/main/scala/zio/http/api/ServiceSpec.scala]: 👍

Comment (afsalthaj) [reply zio-http/src/main/scala/zio/http/api/ServiceSpec.scala]: @jdegoes  Done

Comment (afsalthaj) [reply zio-http/src/main/scala/zio/http/api/ServiceSpec.scala]: The only remaining thing here is to reduce the lengthy compile time error (its better than before IMO) using macros.

Comment (afsalthaj) [reply zio-http/src/main/scala/zio/http/api/ServiceSpec.scala]: Hopefully that could be a separate PR from Kit

**Comment by jdegoes** [comment]
@afsalthaj Can you fix conflicts?

**Comment by afsalthaj** [comment]
Sure

**Comment by afsalthaj** [comment]
@jdegoes  Done!

**Pull request merged**